From 1b8ef484e7a3af4367b771e74c8f72e8326f897f Mon Sep 17 00:00:00 2001
From: Martin Blumenstingl <martin.blumenstingl@googlemail.com>
Date: Fri, 4 Jun 2021 21:50:06 +0200
Subject: [PATCH 61/96] ARM: dts: meson8: Add the shared CMA dma memory pool

The 4K HDMI modes needs more CMA memory (than the default 64MiB) to be
reserved at boot-time. Add a shared-dma-pool with increased size so the
4K HDMI modes can be used.

Signed-off-by: Martin Blumenstingl <martin.blumenstingl@googlemail.com>
---
 arch/arm/boot/dts/amlogic/meson8.dtsi | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/arch/arm/boot/dts/amlogic/meson8.dtsi b/arch/arm/boot/dts/amlogic/meson8.dtsi
index f63ac1404..de9845433 100644
--- a/arch/arm/boot/dts/amlogic/meson8.dtsi
+++ b/arch/arm/boot/dts/amlogic/meson8.dtsi
@@ -193,6 +193,14 @@ power-firmware@4f00000 {
 			reg = <0x4f00000 0x100000>;
 			no-map;
 		};
+
+		linux,cma {
+			compatible = "shared-dma-pool";
+			reusable;
+			size = <0x10000000>;
+			alignment = <0x400000>;
+			linux,cma-default;
+		};
 	};
 
 	thermal-zones {
-- 
2.45.1

